File: Day_16/Q61.cpp
#include <bits/stdc++.h>
using namespace std;

// Input Size
int input_size()
{

    int n;

    do
    {

        cout << "Enter Size : ";
        cin >> n;

        if (n <= 0)
        {
            cout << "Invalid Size" << endl;
        }

    } while (n <= 0);

    return n;
}

// Input Array
void input_array(vector<int> &arr)
{

    cout << "Enter Elements" << endl;

    for (int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

// Finding Missing Number
int find_missing(vector<int> &arr, int n)
{

    int sum = 0;

    int actual_sum = (n * (n + 1)) / 2;

    for (int i = 0; i < arr.size(); i++)
    {
        sum += arr[i];
    }

    return actual_sum - sum;
}

int main()
{

    int n = input_size();

    vector<int> arr(n - 1);

    input_array(arr);

    cout << "Missing Number : " << find_missing(arr, n);

    return 0;
}

File: Day_16/Q61.java
import java.util.Scanner;

class Q61{

    //Input Size
    static int input_size(Scanner sc){

        int n;

        do{

            System.out.print("Enter Size : ");
            n = sc.nextInt();

            if(n <= 0){
                System.out.println("Invalid Size");
            }

        }while(n <= 0);

        return n;

    }

    //Input Array
    static void input_array(int arr[],Scanner sc){

        System.out.println("Enter Elements");

        for(int i = 0;i < arr.length;i++){
            arr[i] = sc.nextInt();
        }

    }

    //Finding Missing Number
    static int find_missing(int arr[],int n){

        int sum = 0;

        int actual_sum = (n*(n+1))/2;

        for(int i = 0;i < arr.length;i++){
            sum += arr[i];
        }

        return actual_sum - sum;

    }

    public static void main(String args[]){

        Scanner sc = new Scanner(System.in);

        int n = input_size(sc);

        int arr[] = new int[n-1];

        input_array(arr,sc);

        System.out.println("Missing Number : " + find_missing(arr,n));

        sc.close();

    }
}